### ISSUE TITLE
[Accessibility]: Code monitors add/edit page not usable on narrow screen

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#35733]
### Audit type

Mobile device navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34195

### Problem description

There are numerous usability issues when trying to create or edit a code monitor on a narrow screen (320px):

- Horizontal padding is too big
- Cards are too narrow with the edit button. Edit button should probably be at the end of the card instead.
- Action cards are overflowing beyond the container
- Trigger query box is so narrow, nothing is visible and (if using codemirror) autocomplete suggestions are not displayed.
- "Preview results" button overflows its container
- Continue/Cancel/Delete buttons inside action cards overflow and wrap in a strange way.
- Slack and webhook alert/help message almost unreadable 

![image](https://user-images.githubusercontent.com/206864/169373306-b70365fa-4f46-4381-aa12-f0ae5884aa08.png)

![image](https://user-images.githubusercontent.com/206864/169373360-218a1782-c83a-4cda-a6d3-8e71f6f04efa.png)

### Expected behavior

Create/edit page has a responsive design that is usable on a narrow screen.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.
